zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -204,28 +407,43 @@ pub fn init_accounts_table<P: consensus::Parameters>(
     // Insert accounts atomically
     wdb.conn.execute("BEGIN IMMEDIATE", NO_PARAMS)?;
     for (account, key) in keys.iter() {
-        let ufvk_str: String = key.encode(&wdb.params);
-        let address_str: String = key.default_address().0.encode(&wdb.params);
-        #[cfg(feature = "transparent-inputs")]
-        let taddress_str: Option<String> = key.transparent().and_then(|k| {
-            k.derive_external_ivk()
-                .ok()
-                .map(|k| k.default_address().0.encode(&wdb.params))
-        });
-        #[cfg(not(feature = "transparent-inputs"))]
-        let taddress_str: Option<String> = None;
-
-        wdb.conn.execute(
-            "INSERT INTO accounts (account, ufvk, address, transparent_address)
-            VALUES (?, ?, ?, ?)",
-            params![<u32>::from(*account), ufvk_str, address_str, taddress_str],
-        )?;
+        add_account_internal(&wdb.params, &wdb.conn, "accounts", *account, key)?;
     }
     wdb.conn.execute("COMMIT", NO_PARAMS)?;
 
     Ok(())
 }
 
+fn add_account_internal<P: consensus::Parameters>(
+    network: &P,
+    conn: &Connection,
+    accounts_table: &'static str,
+    account: AccountId,
+    key: &UnifiedFullViewingKey,
+) -> Result<(), SqliteClientError> {
+    let ufvk_str: String = key.encode(network);
+    let address_str: String = key.default_address().0.encode(network);
+    #[cfg(feature = "transparent-inputs")]
+    let taddress_str: Option<String> = key.transparent().and_then(|k| {
+        k.derive_external_ivk()
+            .ok()
+            .map(|k| k.default_address().0.encode(network))
+    });
+    #[cfg(not(feature = "transparent-inputs"))]
+    let taddress_str: Option<String> = None;
+
+    conn.execute(
+        &format!(
+            "INSERT INTO {} (account, ufvk, address, transparent_address)
+            VALUES (?, ?, ?, ?)",
+            accounts_table
```

Comment:
There's no injection vuln here because `add_account_internal` is private and only called with the strings `accounts` and `accounts_new`.